pkgs/html/lib/src/html_input_stream.dart

```diff
@@ -107,8 +107,9 @@ class HtmlInputStream {
         }
       }
 
-      final isSurrogatePair =
-          i + 1 < charsLength && _isLeadSurrogate(c) && _isTrailSurrogate(rawChars[i + 1]);
+      final isSurrogatePair = _isLeadSurrogate(c) &&
```

Comment:
I would also order it in the original way but when you think about the fact that most characters aren't surrogate, this makes sense as it saves a check per character! Good find!